airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -144,19 +142,21 @@ def create_stream_job(self, query: str, url: str) -> Optional[str]:
         except exceptions.HTTPError as error:
             if error.response.status_code in [codes.FORBIDDEN, codes.BAD_REQUEST]:
                 error_data = error.response.json()[0]
-                if (error_data.get("message", "") == "Selecting compound data not supported in Bulk Query") or (
-                    error_data.get("errorCode", "") == "INVALIDENTITY"
-                    and "is not supported by the Bulk API" in error_data.get("message", "")
+                error_code = error_data.get("errorCode")
+                error_message = error_data.get("message", "")
+                if error_message == "Selecting compound data not supported in Bulk Query" or (
+                    error_code == "INVALIDENTITY" and "is not supported by the Bulk API" in error_message
                 ):
-                    self.logger.error(
-                        f"Cannot receive data for stream '{self.name}' using BULK API, error message: '{error_data.get('message')}'"
-                    )
-                elif error.response.status_code == codes.FORBIDDEN and not error_data.get("errorCode", "") == "REQUEST_LIMIT_EXCEEDED":
-                    self.logger.error(f"Cannot receive data for stream '{self.name}', error message: '{error_data.get('message')}'")
+                    self.logger.error(f"Cannot receive data for stream '{self.name}' using BULK API, error message: '{error_message}'")
+                elif error.response.status_code == codes.FORBIDDEN and error_code != "REQUEST_LIMIT_EXCEEDED":
```

Comment:
can you add some context about why this new logic fixes the problem and what the problem actually is? I don't quite follow how looking at these error codes is related to unqueryable streams

Reply:
added a clarification